File: sdk/typescript/src/api/ngauge.ts
import type { HttpClient } from "../client.js";
import type {
  NGaugeCapacityMetrics,
  NGaugeLeaseList,
  NGaugeListingList,
  NGaugeNodeMetrics,
  NGaugeTrafficMetrics,
} from "../types.js";

export class NGaugeApi {
  constructor(private readonly http: HttpClient) {}

  async capacity(): Promise<NGaugeCapacityMetrics> {
    return this.http.get<NGaugeCapacityMetrics>(
      "/api/v1/ngauge/capacity",
    );
  }

  async traffic(): Promise<NGaugeTrafficMetrics> {
    return this.http.get<NGaugeTrafficMetrics>(
      "/api/v1/ngauge/traffic",
    );
  }

  async marketplaceListings(): Promise<NGaugeListingList> {
    return this.http.get<NGaugeListingList>(
      "/api/v1/ngauge/marketplace/listings",
    );
  }

  async nodeMetrics(): Promise<NGaugeNodeMetrics> {
    return this.http.get<NGaugeNodeMetrics>(
      "/api/v1/ngauge/node/metrics",
    );
  }

  async leases(): Promise<NGaugeLeaseList> {
    return this.http.get<NGaugeLeaseList>("/api/v1/ngauge/leases");
  }
}

File: sdk/cpp/src/api/ngauge.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/ngauge.hpp>

namespace hypermesh {

NGaugeCapacityMetrics NGaugeApi::capacity() const {
    auto j = http_.get("/api/v1/ngauge/capacity");
    return j.get<NGaugeCapacityMetrics>();
}

NGaugeTrafficMetrics NGaugeApi::traffic() const {
    auto j = http_.get("/api/v1/ngauge/traffic");
    return j.get<NGaugeTrafficMetrics>();
}

NGaugeListingList NGaugeApi::marketplace_listings() const {
    auto j = http_.get("/api/v1/ngauge/marketplace/listings");
    return j.get<NGaugeListingList>();
}

NGaugeNodeMetrics NGaugeApi::node_metrics() const {
    auto j = http_.get("/api/v1/ngauge/node/metrics");
    return j.get<NGaugeNodeMetrics>();
}

NGaugeLeaseList NGaugeApi::leases() const {
    auto j = http_.get("/api/v1/ngauge/leases");
    return j.get<NGaugeLeaseList>();
}

} // namespace hypermesh
